find_number.rb

```diff
@@ -0,0 +1,26 @@
+def find_number(number = rand(1..100))
+  puts 'Lets guess a number from 1 to 100.'
+
+  is_guessed = false
+
+  until is_guessed == true
+    puts 'What is your guess?'
+    guess_number = gets.chomp.to_i
+    next puts 'The guess number is not a number!' if guess_number.zero?
+    next puts 'The guess number is out of range 1-100!' unless guess_number.between?(1, 100)
+
+    if guess_number == number
+      puts "Yes, you are right! The number is #{number}"
+      is_guessed = true
+    elsif guess_number > number
+      puts 'Your number is to high.'
+    else
+      puts 'Your number is to low.'
+    end
+  end
+
+  puts "Do you want to play again? Type 'yes' if so, or 'no' if not."
```

Comment:
If we adding checks in the beginning of `find_number.rb` probably a good idea to add checks here as well for consistency.  When running your program, it  does NOT crash with an error which is good but could also be good to provide a warning message if the input is wrong.  Humans are prone to errors sometimes so always good to fortify your code wherever possible.

Reply:
My idea was to check only for `yes` as I'm interested only in it to run a game again. If `no` or anything else is provided - just exit a game.

Checks at the top are other kind of, cause they are inside the game process and is essential part of it.

As for providing a warning message at the end, of course it could be done, but should it be in an infinity cycle until either exactly `yes` or `no` is provided? Doesn't that look a little bit frustrating? 🤔